terraform/module-lambda-roles/main.tf

```diff
@@ -77,7 +78,7 @@ data "aws_iam_policy_document" "payment_reconciler_execution" {
 # Roles
 # ------------------------------------------------------------------------------
 resource "aws_iam_role" "payment_reconciler_execution" {
-  name               = "payment-reconciler-execution"
+  name               = "payment-reconciler-execution-${var.environment}"
```

Comment:
This should probably reference the `service` variable rather than be hardcoded.
I.e. `${var.service}-execution-${var.environment}`

Reply:
Good shout. Done.